reconcile/terraform_vpc_peerings.py

```diff
@@ -32,6 +53,87 @@ def fetch_desired_state(settings):
         }
         peer_connections = peering_info['connections']
         for peer_connection in peer_connections:
+            # We only care about cluster-vpc-requester peering providers
+            if not peer_connection['provider'] == 'cluster-vpc-requester':
+                continue
+
+            # Ensure we have a matching peering connection
+            found, msg = ensure_matching_peering(cluster_info,
+                                                 peer_connection,
+                                                 peer_connection['cluster'],
+                                                 'cluster-vpc-accepter')
+            if not found:
+                return None, msg
+
+            connection_name = peer_connection['name']
+            # peer cluster VPC info
+            peer_vpc_id = peer_connection['cluster']['peering']['vpc_id']
+            peer_vpc_cidr = peer_connection['cluster']['network']['vpc']
+            peer_region = peer_connection['cluster']['spec']['region']
+            # accepter is the target AWS account VPC
+            accepter = {
+                'vpc_id': peer_vpc_id,
+                'cidr_block': peer_vpc_cidr,
+                'region': peer_region,
+            }
+
+            # Find an aws account with the "network-mgmt" access level
+            awsAccount = None
+            for awsAccess in cluster_info['awsInfrastructureAccess']:
+                if awsAccess.get('accessLevel', "") == "network-mgmt":
+                    awsAccount = {
+                        'name': awsAccess['awsGroup']['account']['name'],
+                        'uid': awsAccess['awsGroup']['account']['uid'],
+                        'terraformUsername': (awsAccess['awsGroup']
+                                                       ['account']
+                                                       ['terraformUsername']),
+                    }
+            if not awsAccount:
+                msg = "could not find an AWS account with the " \
+                      "'network-mgmt' access level on the cluster"
+                return None, msg
+
+            # find role to use for aws access
+            awsAccount['assume_role'] = \
+                ocm.get_aws_infrastructure_access_terraform_assume_role(
+                    cluster,
+                    awsAccount['uid'],
+                    awsAccount['terraformUsername']
+                )
```

Comment:
Is there a possibility that this returns empty? It shouldn't be the case as we should always have a `terraform` user in the cluster's access list... Maybe too defensive to check for it, just worth mentioning :)

Reply:
The function should always return a string according to https://github.com/app-sre/qontract-reconcile/blob/ba004224abea6d3b0834a4a353c9873118957a96/utils/ocm.py#L165-L167

Reply:
well that `return` is inside a for loop with two  conditions with `continue` statements. I guess there's a chance that it doesn't return anything...

Reply:
I see what you mean! I've added a test to check if returned value is None (if it never reaches the return statement)